addons/pos_restaurant/static/src/app/services/pos_store.js

```diff
@@ -314,7 +314,13 @@ patch(PosStore.prototype, {
             [...el.classList].find((c) => c.includes("tableId")).split("-")[1]
         );
     },
-    async transferOrder(orderUuid, destinationTable) {
+    getMainTable(table) {
+        while (table.parent_id) {
+            table = table.parent_id;
+        }
+        return table;
+    },
```

Comment:
move this to `restaurant_table.js`.

it should be:

```js
get rootTable() {
    let table = this;
    while (table) {
        table = table.parent_id;
    }
    return table;
}
```

Reply:
I think you can also extract `getActiveOrdersOnTable` and move it to `restaurant_table.js` model.

```js
get activeOrders() {
    return this["<-pos.order.table_id"].filter((o) => !o.finalized && o.lines.length > 0);
}
```

Reply:
with this way When changing the order_id of an orderline in the pos.order.line model, the changes were not reflected correctly. To resolve this, need to add the this line: 
`mergedOrder = this.models["pos.order"].get(mergedOrder.id);`
 to ensure the updated mergedOrder is retrieved
 
 ref: https://github.com/odoo/odoo/pull/188183/files#diff-d2746a4b66902224e35c430b3dcbd8b42124bb03213d750172cb020860f4324dR415

Reply:
> with this way When changing the order_id of an orderline in the pos.order.line model, the changes were not reflected correctly. To resolve this, need to add the this line:
mergedOrder = this.models["pos.order"].get(mergedOrder.id);
to ensure the updated mergedOrder is retrieved

I see. I'm working on a branch that actually resolves such issue. This is a bug in the related_models module.

Reply:
so should we keep this method as it is for now ?

Reply:
yeah, we can keep them.